pages/administration/productions/[productionSlug]/bookings/[bookingReference]/index.vue

```diff
@@ -109,6 +140,37 @@ export default defineNuxtComponent({
         ['Admin Discount', this.rawBooking.adminDiscountPercentage * 100 + '%']
       ];
     }
+  },
+  methods: {
+    anyTicketsChecked(booking) {
+      for (const ticket of booking.tickets) {
+        if (ticket.checkedIn) {
+          return true;
+        }
+      }
+      return false;
```

Comment:
The function can be made into a Vue computed property (https://vuejs.org/guide/essentials/computed.html) as booking is already included in the state (also means less function calls = better performance!). As a general rule, you shouldn't pass state (i.e. things defined in the component's `data` or `props`, such as `booking` here) to functions or methods inside the file too. They can instead use `this.<var>` to refer to it.

JS also has really nice stuff pre-baked for arrays and objects. We can use the `.some()` function on any list to perform the same logic:
```js
return this.booking.tickets.some(ticket => ticket.checkedIn)
```

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
I was hunting for something that does exactly what `.some()` does: glad it exists!